lib/ex_unit/lib/ex_unit/capture_log.ex

```diff
@@ -93,16 +93,17 @@ defmodule ExUnit.CaptureLog do
 
       assert result == 4
       assert log =~ "log msg"
+
   """
   @doc since: "1.13.0"
-  @spec with_log(keyword, (-> any)) :: {any, String.t()}
-  def with_log(opts \\ [], fun) do
+  @spec with_log(keyword, (-> result)) :: {result, String.t()} when result: any
+  def with_log(opts \\ [], fun) when is_list(opts) do
     opts = Keyword.put_new(opts, :level, nil)
     {:ok, string_io} = StringIO.open("")
 
     try do
       :ok = add_capture(string_io, opts)
```

Comment:
We should remove this capture. We are likely capturing everything twice right now. :) plus, we need to pass the level to the ets table and only write if it matches the ETS table level.

Reply:
Note we can default the level to :all.